### PR TITLE
Add #KODIPROP and #EXTVLCOPT to mru playlist to let the user set properties in .strm …

### PR DESCRIPTION
**Pull request by ghost**
…files.. See description.

### CODE REVIEW

**Comment by hudokkow** [comment]
@mapfau, I can't find any information about #EXTPROP reading the [IETF draft] (https://tools.ietf.org/html/draft-pantos-http-live-streaming-18) or anywhere else, for that matter. What's it good for?

Off-topic but can I get you interested in adding support for another property? Problem is described in the [forum] (http://forum.kodi.tv/showthread.php?tid=213527) and is a regular feature request from users.
Should you decide its worth your time, I can give you access to a backend that manifests the problem with some channels.

**Comment by mapfau** [comment]
@hudokkow background is, that inputstream addon can be controlled with CFileItem properties:
1.) inputstreamaddon=inputstream.xyz forces the binary inputstream addon xyz to handle a given stream

2.) in addon.xml of each inputstream addon can be declared a list of accepted properties, pls see here:
https://github.com/mapfau/inputstream.mpd/blob/master/inputstream.mpd/addon.xml.in#L15

This extension here gives strm (and m3u) files the property to a) force an inputstream addon and b) pass values to the addon wich may necessary for it to work correctly.

Nowdays kodi properties already can be set in addons using ListItem::SetProperty(key/value)
This can be done in the same manner with .strm files.

(I'll lokk into the forum thread now,,,)

**Comment by mapfau** [comment]
Would be really no problem to implement the VLC property requested in the thread.
Could be interesting how to pass it to DVDPlayer - @fernetmenta?

**Comment by mapfau** [comment]
The second commit could solve http://forum.kodi.tv/showthread.php?tid=213527
Unfortunateley I have no background on "programs". I'm just guessing, that AVProgram::program_nr ist the thing requested here.

File: xbmc/playlists/PlayListM3U.cpp [hunk 1]
```diff
@@ -43,10 +45,10 @@ const char* CPlayListM3U::OffsetMarker = "#EXT-KX-OFFSET";
 //   #EXTM3U
 //   #EXTART:Demo Artist
 //   #EXTALB:Demo Album
+//   #EXTPROP:name,value
```
Comment (wsnipex) [inline xbmc/playlists/PlayListM3U.cpp]: KODIPROP

Comment (mapfau) [inline xbmc/playlists/PlayListM3U.cpp]: @wsnipex: You are an eagle and not a human :-) Thx.

File: xbmc/playlists/PlayListM3U.cpp [hunk 2]
```diff
@@ -125,6 +129,21 @@ bool CPlayListM3U::Load(const std::string& strFileName)
...
+    || StringUtils::StartsWith(strLine, VLCOptMarker))
+    {
+      size_t iColon = strLine.find(":");
+      size_t iEqualSign = strLine.find("=");
+      if (iColon != std::string::npos &&
+        iEqualSign != std::string::npos &&
+        iEqualSign > iColon)
+      {
+        properties.push_back(std::make_pair(
+          StringUtils::Trim(strLine.substr(iColon, iEqualSign - iColon)),
```
Comment (wsnipex) [inline xbmc/playlists/PlayListM3U.cpp]: travis fails with:
PlayListM3U.cpp:142:72: error: no matching function for call to ‘StringUtils::Trim(std::basic_string<char>)’

           StringUtils::Trim(strLine.substr(iColon, iEqualSign - iColon)),

                                                                        ^

PlayListM3U.cpp:142:72: note: candidates are:

In file included from /home/travis/build/xbmc/xbmc/xbmc/music/Artist.h:28:0,

                 from /home/travis/build/xbmc/xbmc/xbmc/music/Album.h:29,

                 from /home/travis/build/xbmc/xbmc/xbmc/music/tags/MusicInfoTag.h:32,

                 from PlayListM3U.cpp:29:

/home/travis/build/xbmc/xbmc/xbmc/utils/StringUtils.h:73:23: note: static std::string& StringUtils::Trim(std::string&)

   static std::string& Trim(std::string &str);

**Comment by mapfau** [comment]
@wsnipex, thanx - fixed

**Comment by mapfau** [comment]
do you have an slack accont?
markus.pfau@peakwork.com is mine.

**Comment by mapfau** [comment]
@hudokkow could you pls try again?

**Comment by hudokkow** [comment]
Good news! No longer crashes but it doesn't play after file analysis. So... progress I would say.
You have a slack PM with files and credentials.

**Comment by hudokkow** [comment]
Tested and confirmed working! :+1:

**Comment by mapfau** [comment]
beside https://github.com/xbmc/xbmc/pull/9469 (binary post data) this is the second one wich wold make my life really much easier while testing new inputstreamaddons.

I'm not sure if the appVeyour fail is still actual, it builds fine on my machine and to be honest: I cannot interpret appveyor message - hopefully its simply old.

Thanks for looking over it, its only a few lines of code, would be glad if it finds it way into master.

**Comment by FernetMenta** [comment]
@mapfau unfortunately this has missed the merge window that is is from 1st-20th in alpha and beta stage. Only fixes can go in outside this window.
jenkins build this please

**Comment by MartijnKaijser** [comment]
@mapfau add yourself a reminder to ping for merge after 1st May

**Comment by MartijnKaijser** [comment]
@FernetMenta can be merged now i guess?

**Comment by FernetMenta** [comment]
I think so too.

**Pull request merged**